optuna_integration/botorch/botorch.py

```diff
@@ -235,6 +236,11 @@ def qei_candidates_func(
     if train_obj.size(-1) != 1:
         raise ValueError("Objective may only contain single values with qEI.")
     if train_con is not None:
+        if version.parse(botorch.version.version) < version.parse("0.9.0"):
```

Comment:
An error is only raised for constrained problems, since botorch < 0.9.0 will still work fine for unconstrained problems with this change.